tests/includes/juju.sh

```diff
@@ -36,44 +37,41 @@ ensure() {
 	bootstrap "${model}" "${output}"
 }
 
-# bootstrap will attempt to bootstrap a controller on the correct provider.
+# bootstrap will attempt to bootstrap a controller on the correct cloud.
 # It will check if there is an existing controller with the same name and bail,
 # if there is.
 #
 # The name of the controller is randomised, but the model name is used to
 # override the default model name for that controller. That way we have a
-# unqiue namespaced models instead of the "default" model name.
+# unique namespaced models instead of the "default" model name.
 # This helps with providing encapsulated tests without having to bootstrap a
 # controller for every test in a suite.
 #
 # The stdout of the file can be piped to an optional output file.
 #
 # ```
-# bootstrap <model name> <file to output logs>
+# bootstrap <cloud name> <controller name> <file to output logs> <model name>
 # ```
 bootstrap() {
-	local provider name output model bootstrapped_name
+	local cloud name output model bootstrapped_name
 	case "${BOOTSTRAP_PROVIDER:-}" in
 	"aws" | "ec2")
-		provider="aws"
+		cloud="aws"
 		;;
 	"localhost" | "lxd")
-		provider="lxd"
+		cloud="lxd"
 		;;
-	"lxd-remote")
-		provider="lxd-remote"
+	"lxd-remote" | "vsphere" | "openstack"  | "k8s" )
+    cloud="${BOOTSTRAP_CLOUD}"
 		;;
 	"manual")
 		manual_name=${1}
 		shift
 
-		provider="${manual_name}"
-		;;
-	"microk8s")
-		provider="microk8s"
+		cloud="${manual_name}"
 		;;
 	*)
-		echo "Unexpected bootstrap provider (${BOOTSTRAP_PROVIDER})."
+		echo "Unexpected bootstrap cloud (${BOOTSTRAP_PROVIDER})."
```

Comment:
This seems confused that we would set unknown cloud but the  variable is PROVIDER.